frontend/src/components/SignUp.tsx

```diff
@@ -1,20 +1,47 @@
-import { Label } from "@radix-ui/react-label";
-import { Button } from "./ui/button";
+import { Button } from "@/components/ui/button";
 import {
   Card,
+  CardContent,
+  CardDescription,
   CardHeader,
   CardTitle,
-  CardDescription,
-  CardContent,
-} from "./ui/card";
-import { Input } from "./ui/input";
-import { useForm, SubmitHandler, FieldValues } from "react-hook-form";
+} from "@/components/ui/card";
+import {
+  Form,
+  FormControl,
+  FormField,
+  FormItem,
+  FormLabel,
+  FormMessage,
+} from "@/components/ui/form";
+import { Input } from "@/components/ui/input";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { useForm } from "react-hook-form";
 import { Link } from "react-router";
+import { z } from "zod";
+
+const formSchema = z
+  .object({
+    email: z.string().email(),
+    password: z.string().min(8),
+    "repeat-password": z.string().min(8),
```

Comment:
as soon as we have a confirmation from the API, we also need to update here to include the `max`